File: kits/ts/simple/lux/City.ts
// all data related to a city
import {CityTile} from "./CityTile";

export class City {
  public cityid: string;
  public team: number;
  public fuel: number;
  public citytiles = new Array<CityTile>();
  public lightUpkeep: number;

  public constructor(teamid: number, cityid: string, fuel: number, lightUpkeep: number) {
    this.cityid = cityid;
    this.team = teamid;
    this.fuel = fuel;
    this.lightUpkeep = lightUpkeep;
  }

  public addCityTile(x: number, y: number, cooldown: number): CityTile {
    const ct = new CityTile(this.team, this.cityid, x, y, cooldown);
    this.citytiles.push(ct);
    return ct;
  }

  public getLightUpkeep(): number {
    return this.lightUpkeep;
  }
}

File: kits/csharp/simple/Bot/Lux/City.cs
using System.Collections.Generic;

namespace Bot.Lux
{
	public class City
	{
		public string CityId;
		public int Team;
		public double Fuel;
		public List<CityTile> CityTiles = new List<CityTile>();
		private readonly double _lightUpKeep;
		
		public City(int teamId, string cityId, double fuel, double lightUpKeep)
		{
			CityId = cityId;
			Team = teamId;
			Fuel = fuel;
			_lightUpKeep = lightUpKeep;
		}

		public CityTile AddCityTile(int x, int y, double cooldown)
		{
			var ct = new CityTile(Team, CityId, x, y, cooldown);
			CityTiles.Add(ct);
			return ct;
		}

		public double GetLightUpkeep()
		{
			return _lightUpKeep;
		}
    }
}
